File: javascript/interview/countingValleys.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Please enter n : ', (answer1) => {
    rl.question('Please enter x : ', (answer2) => {
        var result = countValleys(+answer1, answer2);
        console.log(`Total number of valleys are:  ${result}`);
        rl.close();
    });
});



function countValleys(n, s) {
  var lvl = 0;
  var v = 0;

  for(var i=0;i < s.length; i++){
    if (s.charAt(i) == 'U')
      lvl+=1;
    if (s.charAt(i) == 'D')
      lvl-=1;
    if (lvl == 0 && s.charAt(i) == 'U')
      v+=1;
  }
  return v;
}

File: R/interview/countingValleys.r
countValleys <- function(n, s){
  lvl = 0
  v = 0

  charString = strsplit(s, "")[[1]]

  for (c in charString){
      if (c == 'U'){
          lvl <- lvl+1
        }
      if (c == 'D'){
          lvl <- lvl-1
        }
      if (lvl == 0 && c == 'U'){
          v <- v + 1
        }
  }
  return(v)
}

main<- function(){
  cat("enter n : ")
  n <- as.integer(readLines("stdin", n=1))
  cat("enter s : ")
  s <- readLines("stdin", n=1)
  result <- countValleys(n, s)
  print(result)
}

main()
